iceoryx_hoofs/legacy/include/iceoryx_hoofs/cxx/expected.hpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_CXX_EXPECTED_HPP
+#define IOX_HOOFS_CXX_EXPECTED_HPP
+
+#include "iox/expected.hpp"
+
+namespace iox
+{
+/// @todo iox-#1593 Deprecate include
+/// [[deprecated("Deprecated in 3.0, removed in 4.0, please include 'iox/expected.hpp' instead")]]
```

Comment:
I'm wondering a bit when this deprecation warning would be shown. When the `cxx` namespace will be used? In this case the deprecation message would be wrong

Reply:
@elBoberido I tried this and had a look very early on when we started with #1391. This will only deprecated what is in exactly this very namespace definition and not everything in `cxx` :D

Reply:
Okay, tried it out. Deprecating namespaces is a C++17 feature and would indeed print it everywhere where the namespace is used. I think we have to add this comment to the respective using directive :(

Reply:
@elBoberido Odd, I tried that :thinking: According to cppref it's C++14: https://en.cppreference.com/w/cpp/language/attributes/deprecated

Reply:
Ah, I might know why it didn't work for you. IIRC you have to write:

```
namespace [[deprecated]] foo
{
  // bla..
}
```

Reply:
I get this warning from my IDE `Attributes on a namespace declaration are a C++17 extension [-Wc++17-extensions]` and when compiling with clang but not with gcc